### ISSUE TITLE
.eval mode predicts [2, -2, 2]

### ISSUE THREAD

**Issue by sonnyhu** [ThibaultGROUEIX/3D-CODED#3]
Hi,

During training, when it runs on validation dataset, it does `network.eval()`. However, I found that whenever it turns to eval mode, the mode predicts `[2, -2, 2]`. And when it stays in train mode, prediction works fine. Did you run into this problem before? Thanks!

I am using default batch size 32 because it seems only layer that will be affected by `eval` is `BatchNorm` which is related to batch size.

**Comment by ThibaultGROUEIX** [comment]
hum that's strange indeed. does the eval script work if you set the network in train mode ? ( and keep batch size of 32 ). that's a good starting point... i never had this issue.

**Comment by sonnyhu** [comment]
Yeah. train mode + 32 batch size works. Based on discussion here (https://github.com/pytorch/pytorch/issues/4741), it seems training is unstable. Not sure what to do... I am going to close issue. If you have some suggestions, definitely let me know. Thanks!

**Comment by sonnyhu** [comment]
Hi again,

I just realized that I was using `0.4.1` pytorch.

For what you said here, "Please note that the Chamfer Distance code doesn't work on all versions of pytorch because of some weird error with the batch norm layers.", does chamfer distance use batch norm or is it in encoder-decoder?

**Comment by ThibaultGROUEIX** [comment]
hi, yes i strongly recomment using the pytorch 0.4 specific commit mentionned in the readme. i am currently porting atlasnet to pytorch 1. i will do 3d-coded after...

**Comment by sonnyhu** [comment]
I tried the commit you listed in readme. Same behavior. Val loss exploded.
![screen shot 2018-11-15 at 12 12 57 pm](https://user-images.githubusercontent.com/4804594/48573662-eb1f1f00-e8da-11e8-98e1-9da203ea6365.png)

**Comment by sonnyhu** [comment]
What should `opt.nb_primitives` be? There is no default value for it. https://github.com/ThibaultGROUEIX/3D-CODED/blob/master/training/train_sup.py#L142

I am looking for any possible way I am training it wrong. Thanks!

**Comment by ThibaultGROUEIX** [comment]
should be 1. but i think these variables are never used anyway.
